### PR TITLE
Additions + Updates pid_data.py

### PR DESCRIPTION
**Pull request by bartvanstiphout**
Added Manufacturers:
- Ayrton
- PRG

Updates
- Clay Paky
- Martin

### CODE REVIEW

File: data/pid_data.py [hunk 1]
```diff
@@ -1148,6 +1206,16 @@
...
 
+      {'get_request': {'items': []},
+       'get_response': {'items': [{'name': 'enabled', 'type': 'uint16'}]},
+       'get_sub_device_range': 0,
+       'name': 'FIXTURE_ID',
+       'set_request': {'items': [{'name': 'enabled', 'type': 'uint16'}]},
+       'set_response': {'items': []},
+       'set_sub_device_range': 0,
+       'link': '',
+       'notes': 'Controls if the fixture will treat key seven on the MCX as a lamp power-off button.',
```
Comment (nomis52) [inline data/pid_data.py]: This notes field looks like it was copied from the PID above and doesn't match the PID name.

Comment (bartvanstiphout) [inline data/pid_data.py]: Should be ok now..

Mvg,

Bart van Stiphout

Mobile NL: +31648560121
Skype: bartvstiphout

www.bartvanstiphout.com <http://www.bartvanstiphout.com/>


	


Now available in the App Store:
Fixture Tester for iPhone/iPad/iPod Touch!
Find out more! <http://itunes.apple.com/us/app/fixture-tester/id368100165?mt=8>

File: data/pid_data.py [hunk 2]
```diff
@@ -2498,6 +2566,214 @@
...
+       'set_request': {'items': [{'name': 'tilt_lock', 'type': 'bool'}]},
+       'set_response': {'items': []},
+       'set_sub_device_range': 0,
+       'link': '',
+       'value': 34305},
+			 
+	  {'get_request': {'items': []},
+       'get_response': {'items': [{'name': 'free_motion', 'type': 'uint8',
+								  'labels': [(1, 'Normal Operation'),
+											 (2, 'Pan Tilt Free Motion'),
```
Comment (peternewman) [inline data/pid_data.py]: Can you fix the spacing so it lines up with the ( above please.

File: data/pid_data.py [hunk 3]
```diff
@@ -2498,6 +2566,214 @@
...
+											 (3, 'Pan Tilt Locked')],
+								  }]},
+       'get_sub_device_range': 0,
+       'name': 'PAN_TILT_FREE_MOTION',
+       'set_request': {'items': [{'name': 'free_motion', 'type': 'uint8',
+								 'labels': [(1, 'Normal Operation'),
+											(2, 'Pan Tilt Free Motion'),
```
Comment (peternewman) [inline data/pid_data.py]: And again please.

File: data/pid_data.py [hunk 4]
```diff
@@ -2498,6 +2566,214 @@
...
+								 }]},
+       'set_response': {'items': []},
+       'set_sub_device_range': 0,
+       'link': '',
+       'value': 34306},
+			 
+	  {'get_request': {'items': []},
+       'get_response': {'items': [{'name': 'followspot_control', 'type': 'uint8',
+								  'labels': [(0, 'Normal Operation'),
+											 (1, 'Intensity Control'),
```
Comment (peternewman) [inline data/pid_data.py]: And Again.

File: data/pid_data.py [hunk 5]
```diff
@@ -2498,6 +2566,214 @@
...
+			 
+	  {'name': 'CLEAR_LOG',
+       'set_request': {'items': [{'name': 'clear_log', 'type': 'bool'}]},
+       'set_response': {'items': []},
+       'set_sub_device_range': 0,
+       'link': '',
+       'value': 34400},
+			 
+	  {'get_request': {'items': []},
+       'get_response': {'items': [{'name': 'diable_mechanics', 'type': 'uint8'}]},
```
Comment (peternewman) [inline data/pid_data.py]: SPaG disable

File: data/pid_data.py [hunk 6]
```diff
@@ -2498,6 +2566,214 @@
...
+       'get_sub_device_range': 0,
+       'name': 'DISABLE_MECHS',
+       'set_request': {'items': [{'name': 'diable_mechanics', 'type': 'uint8'}]},
```
Comment (peternewman) [inline data/pid_data.py]: Again

File: data/pid_data.py (hunk 2 above)
Comment (bartvanstiphout) [inline data/pid_data.py]: Which editor do you use? Mine is constantly fighting your alignments..

Mvg,

Bart van Stiphout

Comment (peternewman) [inline data/pid_data.py]: Nano and do it manually. :) Which editor do you use?

Comment (bartvanstiphout) [inline data/pid_data.py]: Xcode. It aren't tab offset but tabs with some extra spaces. When I press return, I need to re-align it.

Mvg,

Bart van Stiphout

Comment (bartvanstiphout) [inline data/pid_data.py]: Did a new commit.

FYI, this is how it looks in my xcode:


Mvg,

Bart van Stiphout

Mobile NL: +31648560121
Skype: bartvstiphout

www.bartvanstiphout.com <http://www.bartvanstiphout.com/>


	


Now available in the App Store:
Fixture Tester for iPhone/iPad/iPod Touch!
Find out more! <http://itunes.apple.com/us/app/fixture-tester/id368100165?mt=8>

Comment (peternewman) [inline data/pid_data.py]: Yes, we don't use tabs at all. Although FWIW I think the tab equivalent would be two spaces.

If/when you get a working editor config, you should add it here: https://github.com/OpenLightingProject/editor-configs

**Comment by nomis52** [comment]
https://github.com/OpenLightingProject/rdm-app/pull/23 adds pull request validation and fixes the issues you found here.

**Comment by nomis52** [comment]
I've pull this in as #33  and fixed some other formatting issues.

I've left some comments on the PID data that looks a bit odd.

File: data/pid_data.py [hunk 7]
```diff
@@ -126,7 +126,35 @@
        'value': 0x8005},
       ]
   },
-
+  # Ayrton
+  {'id': 16761,
+   'name': 'Ayrton',
+   'pids': [ { 'get_request': { 'items': []},
+               'get_response': { 'items': [{ 'name': 'dimmer_mode', 'type': 'bool'}]},
```
Comment (nomis52) [inline data/pid_data.py]: Using a bool is odd. What's the difference between DIMMER_MODE on vs off? Should this be labels values instead?

File: data/pid_data.py [hunk 8]
```diff
@@ -126,7 +126,35 @@
...
+               'get_sub_device_range': 2,
+               'name': 'DIMMER_MODE',
+               'set_request': { 'items': [{ 'name': 'dimmer_mode', 'type': 'bool'}]},
+               'set_response': { 'items': []},
+               'set_sub_device_range': 1,
+               'value': 32769},
+             { 'get_request': { 'items': []},
+               'get_response': { 'items': [{ 'name': 'light_mode', 'type': 'bool'}]},
+               'get_sub_device_range': 2,
+               'name': 'LIGHT_MODE',
```
Comment (nomis52) [inline data/pid_data.py]: same comment here.

File: data/pid_data.py [hunk 9]
```diff
@@ -126,7 +126,35 @@
...
+               'set_request': { 'items': [{ 'name': 'light_mode', 'type': 'bool'}]},
+               'set_response': { 'items': []},
+               'set_sub_device_range': 1,
+               'value': 32770},
+             { 'get_request': { 'items': []},
+               'get_response': { 'items': [{ 'name': 'constant_color_modes', 'type': 'bool'}]},
+               'get_sub_device_range': 2,
+               'name': 'CONSTANT_COLOR_MODES',
```
Comment (nomis52) [inline data/pid_data.py]: and here.

File: data/pid_data.py [hunk 10]
```diff
@@ -2498,6 +2566,214 @@
...
+											 (4, '300ft Throw'),
+											 (5, '15ft Throw'),
+								 }]},
+       'set_response': {'items': []},
+       'set_sub_device_range': 0,
+       'link': '',
+       'value': 34337},
+			 
+	  {'name': 'ZERO_GOBO_WHEELS',
+       'set_request': {'items': [{'name': 'followspot_control', 'type': 'bool'}]},
```
Comment (nomis52) [inline data/pid_data.py]: The field name appears to be wrong.

File: data/pid_data.py [hunk 11]
```diff
@@ -2498,6 +2566,214 @@
...
+			 
+	  {'name': 'CLEAR_LOG',
+       'set_request': {'items': [{'name': 'clear_log', 'type': 'bool'}]},
+       'set_response': {'items': []},
+       'set_sub_device_range': 0,
+       'link': '',
+       'value': 34400},
+			 
+	  {'get_request': {'items': []},
+       'get_response': {'items': [{'name': 'disable_mechanics', 'type': 'uint8'}]},
```
Comment (nomis52) [inline data/pid_data.py]: Should this be a bool ?

Comment (bartvanstiphout) [inline data/pid_data.py]: All this info I've got from the fixtures itself, not from documents. For some values I don't have an explanation, but in my opinion it is better to have something then nothing.

Mvg,

Bart van Stiphout

**Pull request merged**